substrate/client/rpc/src/state/state_full.rs

```diff
@@ -406,24 +405,33 @@ where
 			});
 
 		let stream = futures::stream::once(future::ready(initial)).chain(version_stream);
-
-		let fut = async move {
-			sink.pipe_from_stream(stream).await;
-		};
-
-		self.executor.spawn("substrate-rpc-subscription", Some("rpc"), fut.boxed());
+		spawn_subscription_task(&self.executor, pipe_from_stream(pending, stream, 16));
 	}
 
-	fn subscribe_storage(&self, mut sink: SubscriptionSink, keys: Option<Vec<StorageKey>>) {
+	fn subscribe_storage(
+		&self,
+		pending: PendingSubscriptionSink,
+		keys: Option<Vec<StorageKey>>,
+		deny_unsafe: DenyUnsafe,
+	) {
+		if keys.is_none() {
+			if let Err(err) = deny_unsafe.check_if_safe() {
+				spawn_subscription_task(&self.executor, pending.reject(ErrorObject::from(err)));
```

Comment:
Does this change the behavior of the `subscribe_storage`?
We previously rejected the subscription on `self.client.storage_changes_notification_stream` failures. Was this added as an optimization here that would have been otherwise reported by `storage_changes_notification_stream`?

Reply:
To answer this: `state/mod.rs` did the same thing, I believe this is entirely fine